pygments/lexers/dotnet.py

```diff
@@ -704,3 +704,29 @@ def analyse_text(text):
             result += 0.05
 
         return result
+
+
+class BlazorLexer(RegexLexer):
+    """
+    For `Blazor <https://docs.microsoft.com/en-us/aspnet/core/blazor/?view=aspnetcore-5.0>`_
+    source code.
+    """
+
+    name = 'Blazor'
+    aliases = ['blazor']
+    filenames = ['*.razor']
+    mimetypes = []
+
+    tokens = {
+        'root': [
+            (r'(@code |@page )', Keyword),
```

Comment:
You normally don't want the space as part of the keyword. To achieve that, this rule should read

```python
(r'(@code|@page)( )', bygroups(Keyword, Whitespace))
```

Also, isn't `@code{` allowed?